main.cpp

```diff
@@ -47,11 +47,55 @@ struct Point
 template<typename Type>
 struct Wrapper
 {
-    Wrapper(Type&& t) : val(std::move(t)) 
+    Wrapper(Type&& t) : val(std::move(t))
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")\n";
+    }
+
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+
+    using Type = Point;
+
+    Wrapper(Type&& t) : val(std::move(t))
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")\n"; //call Point's toString function
+    }
+
+    Type val; //in this case, this holds an instance of Point
+};
+
+template<typename T>
+void variadicHelper(T&& first)
+{
+    //Do something with 'first' here: Instruction 6
+    Wrapper wrapper { std::forward<T>(first) };
+    wrapper.print();
+
+    //variadicHelper( std::forward<Args>(everythingElse) ... ); //recursive call
+}
+
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... everythingElse)
+{
+    //Do something with 'first' here: Instruction 6
+    Wrapper wrapper { std::forward<T>(first) };
+    wrapper.print();
+
+    variadicHelper( std::forward<Args>(everythingElse) ... ); //recursive call
+}

```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?